app/src/androidTest/java/com/github/se/eduverse/ui/folder/ListFoldersTest.kt

```diff
@@ -145,4 +151,65 @@ class ListFoldersTest {
+    composeTestRule.onNodeWithTag("delete").assertIsNotDisplayed()
+    composeTestRule.onNodeWithTag("cancel").assertIsNotDisplayed()
+    composeTestRule.onAllNodesWithTag("checked").assertCountEquals(0)
+    composeTestRule.onAllNodesWithTag("unchecked").assertCountEquals(0)
+
+    composeTestRule.onNodeWithTag("folderCard1").performTouchInput { longClick() }
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag("delete").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("cancel").assertIsDisplayed()
+    composeTestRule.onAllNodesWithTag("checked").assertCountEquals(1)
+    composeTestRule.onAllNodesWithTag("unchecked").assertCountEquals(1)
+
+    composeTestRule.onNodeWithTag("checked").performClick()
+    composeTestRule.onNodeWithTag("delete").assertIsDisplayed()
+    composeTestRule.onAllNodesWithTag("checked").assertCountEquals(0)
+    composeTestRule.onAllNodesWithTag("unchecked").assertCountEquals(2)
+
+    composeTestRule.onNodeWithTag("cancel").performClick()
+    composeTestRule.onNodeWithTag("delete").assertIsNotDisplayed()
+    composeTestRule.onNodeWithTag("cancel").assertIsNotDisplayed()
+    composeTestRule.onAllNodesWithTag("checked").assertCountEquals(0)
+    composeTestRule.onAllNodesWithTag("unchecked").assertCountEquals(0)
+
+    composeTestRule.onNodeWithTag("folderCard1").performTouchInput { longClick() }
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag("delete").assertIsDisplayed()
+    composeTestRule.onAllNodesWithTag("checked").assertCountEquals(1)
+    composeTestRule.onAllNodesWithTag("unchecked").assertCountEquals(1)
+
+    composeTestRule.onNodeWithTag("unchecked").performClick()
+    composeTestRule.onNodeWithTag("delete").assertIsDisplayed()
+    composeTestRule.onAllNodesWithTag("checked").assertCountEquals(2)
+    composeTestRule.onAllNodesWithTag("unchecked").assertCountEquals(0)
+
+    composeTestRule.onNodeWithTag("confirm").assertIsNotDisplayed()
+    composeTestRule.onNodeWithTag("delete").performClick()
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag("confirm").assertIsDisplayed()
+
+    composeTestRule.onNodeWithTag("no").performClick()
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag("confirm").assertIsNotDisplayed()
+    composeTestRule.onAllNodesWithTag("checked").assertCountEquals(2)
+
+    verify(0) { folderRepository.deleteFolder(any(), any(), any()) }
+
+    composeTestRule.onNodeWithTag("delete").performClick()
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag("yes").performClick()
+    composeTestRule.waitForIdle()
+
+    composeTestRule.onNodeWithTag("confirm").assertIsNotDisplayed()
+    composeTestRule.onNodeWithTag("delete").assertIsNotDisplayed()
+    composeTestRule.onAllNodesWithTag("checked").assertCountEquals(0)
+    composeTestRule.onAllNodesWithTag("unchecked").assertCountEquals(0)
+
+    verify(1) { folderRepository.deleteFolder(any(), any(), any()) }
+  }
 }
```

Comment:
The implemented tests are well designed and effectively cover important cases. They accurately check the initial state of components, user interactions, as well as confirmation or cancellation of deletion via dialog. Well done!